scripts/00_training/computer_vision.py

```diff
@@ -0,0 +1,300 @@
+                with torch.no_grad():
+                    for batch in test_dataloader:
+                        x, y = batch
+                        y_onehot = F.one_hot(y, num_classes=num_classes).float()
+                        output = model(x)
+
+                        eval_loss += criterion(output, y_onehot).item()
+
+                        predictions = torch.argmax(output, dim=1)
+                        predictions, y = accelerator.gather_for_metrics((predictions, y))
+                        test_accuracy.add_batch(predictions=predictions, references=y)
+
+                eval_loss = eval_loss / len(test_dataloader) 
+                train_accuracy_metric = train_accuracy.compute()
+                test_accuracy_metric = test_accuracy.compute()
+
+                if use_wandb:
+                    wandb.log(
+                        data={
+                            "train_loss": train_loss, # loss from the latest batch
+                            "eval_loss": eval_loss,
+                            "train_accuracy": train_accuracy_metric,
+                            "eval_accuracy": test_accuracy_metric,
+                        },
+                        step=epoch,
+                    )
+                # custom eval
+                data = get_metrics(model)
+                data['train_loss'] = train_loss
+                data['eval_loss'] = eval_loss
+                data['train_accuracy'] = train_accuracy_metric
+                data['eval_accuracy'] = test_accuracy_metric
+
+                with open(os.path.join(run_output_dir, f"step{steps}.json"), "w") as f:
+                    json.dump(data, f)
+
+                # accelerator.wait_for_everyone()
+                # accelerator.save(model.state_dict(), os.path.join(run_output_dir, f"model_{steps}.pt"))
+                model.train()
+
+    accelerator.wait_for_everyone()
+    accelerator.save(model.state_dict(), os.path.join(run_output_dir, f"model.pt"))
+
+    if wandb:
+        wandb.finish()
+
+
+# %%
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--output_dir", type=str, default="output")
+    parser.add_argument("--seed", type=int, default=0)
+    parser.add_argument("--lr", type=float, default=1e-3)
+    parser.add_argument("--num_epochs", type=int, default=100)
+    parser.add_argument("--input_dim", type=int, default=784)
+    parser.add_argument("--hidden_dims", type=int, nargs="+", default=[800])
+    parser.add_argument("--output_dim", type=int, default=10)
+    parser.add_argument("--init_scaling", type=float, default=1.0)
+    parser.add_argument("--train_bsz", type=int, default=256)
+    parser.add_argument("--train_subsample", type=int, default=1000)
+    parser.add_argument("--cpu", action="store_true")
+    parser.add_argument("--eval_every", type=int, default=20)
+    parser.add_argument("--wandb", action="store_true")
+    parser.add_argument("--dataset_name", type=str, default="mnist")
```

Comment:
Since there are a finite number of choices for `dataset_name`, you should use the argparse `choices` parameter (see https://docs.python.org/3/library/argparse.html#choices). Then you can also delete L110-111 above. It's both better for code readability and error checking.